### PR TITLE
Ignore corrupted tables on `sys.shards` queries

### PR DESCRIPTION
**Pull request by mfussenegger**
I'm not really sure if this is a good idea and whether we still need it after

- https://github.com/crate/crate/pull/10713
- https://github.com/crate/crate/pull/10706

It could potentially mask bugs?

## Checklist

 - [ ] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/main/java/io/crate/execution/engine/collect/sources/ShardCollectSource.java [hunk 1]
```diff
@@ -491,6 +491,8 @@ private ShardCollectorProvider getCollectorProviderSafe(ShardId shardId) {
                     shardRowContexts.add(shardCollectorProvider.shardRowContext());
                 } catch (ShardNotFoundException | IllegalIndexShardStateException e) {
                     unassignedShards.add(toUnassignedShard(index.getName(), shard.value));
+                } catch (Throwable t) {
+                    LOGGER.error("Error retrieving shard information for index={} error={}", index.getName(), t);
```
Comment (mfussenegger) [inline server/src/main/java/io/crate/execution/engine/collect/sources/ShardCollectSource.java:495]: Maybe we could add an additional shard state `corrupted` and add the error message to it. So users can still see the shards for tables that have a corrupt schema?

Comment (seut) [reply server/src/main/java/io/crate/execution/engine/collect/sources/ShardCollectSource.java:495]: > Maybe we could add an additional shard state corrupted and add the error message to it. So users can still see the shards for tables that have a corrupt schema?

Good idea, that would at least give us hints for potential masked bugs here. +1

Comment (mkleen) [reply server/src/main/java/io/crate/execution/engine/collect/sources/ShardCollectSource.java:495]: +1

**Comment by mfussenegger** [comment]
Won't do this for now. Also not https://github.com/crate/crate/pull/10714#discussion_r513319323

Users shouldn't be able to create broken tables or modify tables in a way that breaks them and we should rather fix bugs in that area than trying to cope with broken tables.

`DROP TABLE` can be used as an escape hatch and we already improved the validation for CREATE and ALTER TABLE.

**Pull request closed without merging**